koduyorum-web/src/PageComponents.js

```diff
@@ -44,6 +43,23 @@ export const Navbar = ({
   handleSearchResultClick,
 }) => {
   const navigate = useNavigate();
+  const [username, setUsername] = useState(null);
+
+  const handleProfileClick = async () => {
+    try {
+      const storedUsername = localStorage.getItem('username');
+      
+      if (storedUsername) {
+        setUsername(storedUsername);
```

Comment:
I really dont know if it is the right code or not since I didn't write front-end